packages/shared/src/components/cards/PostMetadata.tsx

```diff
@@ -1,30 +1,29 @@
 import React, { ReactElement, ReactNode, useMemo } from 'react';
 import classNames from 'classnames';
-import { Post } from '../../graphql/posts';
+import { Post, ReadHistoryPost } from '../../graphql/posts';
 import { postDateFormat } from '../../lib/dateFormat';
-import classed from '../../lib/classed';
-
-const Separator = classed(
-  'div',
-  'mx-1 w-0.5 h-0.5 rounded-full bg-theme-label-tertiary',
-);
+import { Separator } from './common';
 
 interface PostMetadataProps {
-  post: Post;
+  createdAt?: string;
```

Comment:
I think the main benefit of using the `Post` interface is that we can get those properties from one place in case they might change.

Up to you, but optionally we could use a `Partial<Post, 'createdAt' | 'readTime' | 'numUpvotes>` type.

Reply:
In this case I was thinking that PostMetadata component could be only responsible to **how** to show the data and the components who are calling PostMetadata are responsible to **what** to show from the Post. WDYT?

Reply:
Hmm I personally think it could give us issues by not leveraging the type already existing.
For instance readTime, let's say it's a number now, but we decide to make it a string because of some reason.

But either way will do really.

Reply:
Chris is right. You should use `Pick<Post, 'createdAt' | 'readTime' | 'numUpvotes'>` to make sure types are aligned throughout the app. You still get what you want @ayadotdev where the caller will be in charge of extracting the properties but it will be aligned with the right types

Reply:
@rebelchris @idoshamun I got your suggestions, it makes sense to align the types through the app. The issue is that some components display only some metadata, e.x. the card component doesn't display numUpvotes but displays createdAt and readTime, while the ReadingHistoryItem displays readTime, numUpvotes but not the createdAt. It means that to manage that logic the caller should pass some flags together with post metadata which seems a bit convoluted. WDYT?

Reply:
I think that's why we should use optional characters.
Which Partial can do, depending on which one is always required?

I can help you set this type up if you want.
The main goal is to have shared types throughout the app, which we currently don't always do.

We could even mix with a Union type actually. So it can either be part of a Post or part of the ReadingHistoryItem?

Reply:
I think what Aya is concerned about is that `post` in the feed will always have `numUpvotes` and it should not be displayed in the card's metadata (which will automatically be displayed if the whole object was passed).

Reply:
Yes, Lee, exactly.

Reply:
Oh right, but that has nothing to do with the Type of the Props?
You can still pass the individual items for each one, so the feed is the only one that passes `numUpvotes`.

We are concerned about the types of `PostMetadataProps`.
Let me know if you want to have a chat about this.